### PR TITLE
toml2nix: init at 0.1.1

### PR DESCRIPTION
**Pull request by P-E-Meunier**
###### Motivation for this change
Toml2nix is a small tool to convert TOML files to Nix expressions. It will be particularly useful to solve the recurring portability problems of https://github.com/mozilla/nixpkgs-mozilla

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5809,6 +5809,8 @@ in
 
   tokei = callPackage ../development/tools/misc/tokei { };
 
+  toml2nix = (callPackage ../tools/toml2nix { }).toml2nix { };
```
Comment (dywedir) [inline pkgs/top-level/all-packages.nix]: `../tools/toml2nix` folder is missing in your PR

**Comment by dywedir** [comment]
@GrahamcOfBorg build toml2nix

**Comment by Mic92** [comment]
There could a function in lib that uses either the `fromToml` builtin or fallback to this program.

**Comment by nbp** [comment]
@Mic92 not within Nixpkgs, as nixpkgs should not use the import-from-derivation pattern.

**Pull request merged**